src/main/java/org/mule/extension/http/internal/request/ShareableHttpClient.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.http.internal.request;
+
+import org.mule.runtime.http.api.client.HttpClient;
+import org.mule.runtime.http.api.client.auth.HttpAuthentication;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Wrapper implementation of an {@link HttpClient} that allows being shared by only configuring the client when first required
+ * and only disabling it when last required.
+ */
+public class ShareableHttpClient {
```

Comment:
what's the need for making this class top level? It doesn't seem used anywhere in the PR

Reply:
It is being used by the wrapper's base connection provider.